arches_for_science/migrations/0005_deploy_triggers_for_primary_descriptor.py

```diff
@@ -0,0 +1,66 @@
x;\n        SELECT\n        CASE\n            WHEN loop_index = 1 THEN 'name'\n            WHEN loop_index = 2 THEN 'map_popup'\n            ELSE 'description'\n        END\n        INTO descriptor_key;\n\n        -- Resolve node values to localized strings\n        FOR alias_with_separators IN SELECT UNNEST(REGEXP_MATCHES(this_template, '<\\w+>', 'g'))\n        LOOP\n            SELECT TRIM(BOTH '<>' FROM alias_with_separators) INTO node_alias;\n\n            SELECT nodeid, nodegroupid INTO node_for_alias, nodegroup_for_alias\n                FROM nodes\n                WHERE alias = node_alias\n                AND graphid = graph;\n\n            SELECT tiledata ->> node_for_alias::text INTO localized_string_node_value\n                FROM tiles\n                WHERE resourceinstanceid = resourceid\n                AND nodegroupid = nodegroup_for_alias\n                AND sortorder = 0\n                LIMIT 1;\n\n            -- Replace template with localized string\n            FOR lang, inner18n_obj_for_lang IN SELECT * FROM jsonb_each(localized_string_node_value) \n            LOOP\n                -- Initialize language key if missing\n                IF localized_result_all -> lang IS NULL THEN\n                    SELECT jsonb_set(\n                        localized_result_all,\n                        ARRAY[lang],\n                        JSONB_OBJECT('{}')\n                    ) INTO localized_result_all;\n                END IF;\n\n                -- Retrieve the current working value, or start from the template\n                SELECT\n                CASE\n                    WHEN localized_result_all -> lang -> descriptor_key IS NULL\n                        THEN this_template\n                    ELSE TRIM((localized_result_all -> lang -> descriptor_key)::text, '\\\"')\n                END\n                INTO working_string;\n\n                SELECT REPLACE(\n                    working_string,\n                    alias_with_separators,\n                    TRIM((inner18n_obj_for_lang -> 'value')::text, '\"')\n                )\n                INTO resolved_node_value_for_lang;\n\n                -- Update the working value\n                SELECT jsonb_set(\n                    localized_result_all,\n                    ARRAY[lang, descriptor_key],\n                    TO_JSONB(resolved_node_value_for_lang)\n                ) INTO localized_result_all;\n\n                IF descriptor_key = 'name' THEN\n                    SELECT jsonb_set(\n                        localized_result_name_only,\n                        ARRAY[lang],\n                        TO_JSONB(resolved_node_value_for_lang)\n                    ) INTO localized_result_name_only;\n                END IF;\n\n            END LOOP;\n\n        END LOOP;\n\n    END LOOP;\n\n    UPDATE resource_instances\n    SET descriptors = localized_result_all, name = localized_result_name_only\n    WHERE resourceinstanceid = resourceid;\n\nEND LOOP;\nEND;\nRETURN NULL;\n",
```

Comment:
The function dump (for this particularly complicated function) is ugly, but it's not a migration we'll be manually editing.